buildchain/dodo.py

```diff
@@ -112,6 +120,11 @@ def _write_failure(self, result: Dict[str, Any]) -> None:
         self._write(result["exception"].get_msg())
         self._write("\n")
 
+    def _write_time_elapsed(self, task: doit.task.Task) -> None:
+        if task.actions:
+            time_elapsed = int(self.t_results[task.name].to_dict()["elapsed"])
```

Comment:
`TaskResult.elapsed` should be of type `Optional[float]` already, so you're using the type cast as an implicit guard against `None` values? Or just to essentially truncate the float value to remove sub-second precision digits? Either way, could be better to be explicit, either with a comment or with more explicit code:

```python
# Use `to_dict` to evaluate the "elapsed" value
task_result = self.t_results[task.name].to_dict()
elapsed = task_result["elapsed"]
assert elapsed is not None, "Some runtime error message"
# `elapsed` is a float, but we strip sub-second digits
self._write(f" [{elapsed:.0f}s]\n")
```